app/client/components/LinkingState.ts

```diff
@@ -9,150 +9,220 @@ import {FilterColValues, QueryOperation} from "app/common/ActiveDocAPI";
-      const _filterColValues = ko.observable<FilterColValues>();
-      this.filterColValues = this.autoDispose(ko.computed(() => _filterColValues()));
-
-      // source data table could still be loading (this could happen after changing the group by
-      // columns of a linked summary table for instance), hence the below listener.
-      this.autoDispose(srcTableData.dataLoadedEmitter.addListener(_update));
+    }));
 
-      _update();
-      function _update() {
-        const result: FilterColValues = {filters: {}, operations: {}};
-        if (srcSection.isDisposed()) {
-          return result;
+    if (srcSection.selectedRowsActive()) { //old, special-cased custom filter
+        const operation = (tgtColId && isRefListType(tgtCol.type())) ? 'intersects' : 'in';
+        this.filterState = this._srcCustomFilter(tgtCol, operation); // works whether tgtCol is the empty col or not
+
+    } else if (tgtColId) { //Standard filter link
+      //If srcCol is the empty col, is a row->col filter (i.e. id -> tgtCol)
+      //else is a col->col filter (srcCol -> tgtCol)
+      //MakeFilterObs handles it either way
+      this.filterState = this._makeFilterObs(srcCol, tgtCol);
+
+    } else if (srcColId && isRefListType(srcCol.type())) {  // "Show Referenced Records" link
+      //tgtCol is the emptycol (i.e. the id col)
+      //srcCol must be a reference to the tgt table
+      //Link will filter tgt section to show exactly the set of rowIds referenced by the srcCol
+      //(NOTE: currently we only do this for reflists, single refs handled as cursor links for now)
+      this.filterState = this._makeFilterObs(srcCol, undefined);
+
+    } else if (!srcColId && isSummaryOf(srcSection.table(), tgtSection.table())) { //Summary linking
+      // We do summary filtering if no cols specified and summary section is linked to a more detailed summary
+      // (or to the summarySource table)
+      // Implemented as multiple column filters, one for each groupByCol of the src table
+
+      // temp vars for _update to use (can't set filterState directly since it's gotta be a computed)
+      const _filterState = ko.observable<FilterState>();
+      this.filterState = this.autoDispose(ko.computed(() => _filterState()));
+
+      //update may be called multiple times, so need a holder to handle disposal
+      //Note: grainjs MultiHolder can't actually be cleared. To be able to dispose of multiple things, we need
+      //      to make a MultiHolder in a Holder, which feels ugly but works.
+      // TODO: Update this if we ever patch grainjs to allow multiHolder.clear()
+      const updateHolder = Holder.create(this);
+
+      // source data table could still be loading (this could happen after changing the group-by
+      // columns of a linked summary table for instance). Define an _update function to be called when data loads
+      const _update = () => {
```

Comment:
It turns out that this whole exercise here (to not subscribe to data) doesn't work and it hasn't worked before. The whole linking state is rebuild when row data is changed, but we can update it later in a future diff.
From what I understand, the whole point of this complicated code is that typings for ko.computed and ko.observable are diffrent (even thouge there is common core and it is typed in grainjs).

Reply:
Yeah, I noticed the row-rebuild issue also. Probably needs to be written with grainjs observables to  be done correctly, probably a problem for a later diff